src/main/java/seedu/address/model/student/Year.java

```diff
@@ -25,17 +26,26 @@
 
     /**
      * Constructs a {@code Year}.
-     *
-     * @param schoolType A valid school type
-     * @param level A valid level
      */
-    public Year(SchoolType schoolType, Integer level) {
-        requireNonNull(schoolType);
-        requireNonNull(level);
+    public Year(String year) {
+        requireNonNull(year);
+        final Matcher matcher = YEAR_FORMAT.matcher(year.trim());
+        assert matcher.matches();
+        String schoolTypeString = matcher.group("school").trim().toLowerCase();
+        String levelString = matcher.group("level").trim().toLowerCase();
+
+        SchoolType schoolType = getSchoolType(schoolTypeString);
+        Integer level = Integer.parseInt(levelString);
         this.schoolType = schoolType;
         this.level = level;
     }
 
+    private static SchoolType getSchoolType(String schoolTypeString) {
```

Comment:
Hmm would this method be better off in `SchoolType.java` instead since it's about `SchoolType`? Also, I think because it's a static method the assert statement can potentially cause bugs if the method is used for other things.

Reply:
It really is just a helper method for the constructor to maintain the abstraction level between `Integer::parseInt` and the SchoolType parsing, and no other class will use it, but no harm doing so I guess.